core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -310,11 +330,52 @@ public void run() {
     }
   }
 
-  private ScheduledFuture<?> startDeadlineTimer(Deadline deadline) {
+  private ScheduledFuture<?> startDeadlineTimer(final Deadline deadline) {
     long remainingNanos = deadline.timeRemaining(TimeUnit.NANOSECONDS);
-    return deadlineCancellationExecutor.schedule(
-        new LogExceptionRunnable(
-            new DeadlineTimer(remainingNanos)), remainingNanos, TimeUnit.NANOSECONDS);
+    try {
+      return deadlineCancellationExecutor
+          .schedule(new LogExceptionRunnable(new DeadlineTimer(remainingNanos)), remainingNanos,
+                    TimeUnit.NANOSECONDS);
+    } catch (RejectedExecutionException e) {
+      class CancelledFuture implements ScheduledFuture<Void> {
+        @Override
+        public long getDelay(TimeUnit unit) {
+          return deadline.timeRemaining(unit);
+        }
+
+        @Override
+        public int compareTo(Delayed delayed) {
+          throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean cancel(boolean mayInterruptIfRunning) {
+          return true;
+        }
+
+        @Override
+        public boolean isCancelled() {
+          return true;
+        }
+
+        @Override
+        public boolean isDone() {
+          return false;
+        }
+
+        @Override
+        public Void get() {
+          throw new CancellationException();
+        }
+
+        @Override
+        public Void get(long timeout, TimeUnit unit) {
+          throw new CancellationException();
+        }
+      }
+
+      return new CancelledFuture();
```

Comment:
I don't feel comfortable silently ignoring the deadline. That seems just asking for undiagnosable run-away RPCs. The graceful way to handle this would be to fail the RPC.